### ISSUE TITLE
[Bug]: Focus shifts outside the dialog when tab key is pressed continously

### ISSUE THREAD

**Issue by Plahot** [microsoft/fluentui#26358]
### Library

React v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz
    Memory: 49.91 GB / 63.75 GB
  Browsers:
    Edge: Spartan (44.22621.1105.0), Chromium (109.0.1518.55)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://react.fluentui.dev/?path=/docs/components-dialog--default 

## Actual Behavior
Focus shifts outside the dialog box when continuously tab key is pressed, and dialog is not closed.
Issue is also present in documentation of Fluent UI React v9: https://react.fluentui.dev/?path=/docs/components-dialog--default

## Expected Behavior
Focus should not shift outside the dialog box and if it does dialog must be closed as described in the modal.

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by layershifter** [comment]
Was able to repro it.

- Open Dialog with keyboard
- Focus is inside `Dialog`
- Press `Tab`, focus goes outside `Dialog` (that's fine as it should return to navbar)
- Press `Tab`, focus goes to Storybook menu (that's fine as stories appear in iframe)
- Press `Tab`, focus goes to a focusable element within an iframe and is **outside** `Dialog`

**Comment by sjwilczynski** [comment]
@layershifter could you give more insights why 
> Press Tab, focus goes outside Dialog (that's fine as it should return to navbar)
> Press Tab, focus goes to Storybook menu (that's fine as stories appear in iframe)
are expected? When I look at the documentation, I understood that focus will be trapped inside the dialog:
![image](https://user-images.githubusercontent.com/19153565/214546725-7e149f42-c403-48eb-afce-52f5c8dc2803.png)

So what's the expected behavior? Can user, just using keyboard move outside of dialog? If so shouldn't we have an option to trap it?

**Comment by bsunderhus** [comment]
The expected behavior is the same as the one present on native elements @sjwilczynski, Focus trapping should only trap focus on the context of the current active tab, and not in the context of the browser.

https://codesandbox.io/s/lively-surf-x03cnr?file=/src/App.tsx

Tabbing should move you out of the application, like in this native `dialog` example.

**Comment by bsunderhus** [comment]
> Press Tab, focus goes to a focusable element within an iframe and is outside Dialog 💥

I wasn't able to reproduce this issue on our documentation @layershifter. Nonetheless if that's the case than this seems to be a possible tabster issue. Needs more investigation.

**Comment by layershifter** [comment]
> I wasn't able to reproduce this issue on our documentation @layershifter. Nonetheless if that's the case than this seems to be a possible tabster issue. Needs more investigation.

> Was able to repro it.
> 
> * Open Dialog with keyboard
> * Focus is inside `Dialog`
> * Press `Tab`, focus goes outside `Dialog` (that's fine as it should return to navbar)
> * Press `Tab`, focus goes to Storybook menu (that's fine as stories appear in iframe)
> * Press `Tab`, focus goes to a focusable element within an iframe and is **outside** `Dialog` 💥

@bsunderhus updated repro steps:

- Open https://react.fluentui.dev/?path=/docs/components-dialog--default
- Tab to "Default" example, ensure that "Open Dialog" button is focused
- A dialog should open, "Close" button should be focused
- Hold <kbd>Tab</kbd> key (i.e. press down)
- Observe that focus goes through the page including elements under a dialog

**Comment by bsunderhus** [comment]
The interesting is that as soon as the tab key is release the focus goes immediately back to the last element inside the modal. I'll have this issue properly discussed and then I'll come back to it! thx for reporting it @Plahot

**Comment by mshoho** [comment]
I believe this is already fixed with latest Tabster 4.1.1 which is already in FUI master.
